content/ch-applications/Variational Quantum Regression.ipynb

```diff
@@ -0,0 +1,893 @@
+   "metadata": {},
+   "source": [
+    "## Translate to Quantum Domain\n",
+    "\n",
+    "We now need to explore how we will translate the data sets, x and y, onto a quantum computer. Let us think of x and y as length N vectors. The easiest way to encode these data sets onto a quantum computer is by initialising qubits in the states $\\ket{x}$ and $\\ket{y}$, where \n",
+    "\n",
+    "$$\\ket{x} = \\frac{1}{C_x}\\vec{x}$$ and $C_x$ is a normalisation factor. Note that all of our starting states must be normalised.\n",
+    "\n",
+    "Due to the definition of the tensor product and the fact that the general statevector of a single qubit is a vector of length 2, $n$ qubits can encode length $2^n$ vectors. \n",
+    "\n",
+    "### Cost Function\n",
+    "\n",
+    "Thus, we propose a trial solution for the two unknown parameters, a and b, as\n",
+    "\n",
+    "$$\\ket{\\Phi} = a\\frac{C_x}{C_y}\\ket{x} + \\frac{b}{C_y}\\ket{1}$$ \n",
+    "\n",
+    "Our proposed cost function, which we wish to minimise is equal to\n",
+    "\n",
+    "$$C_P = \\left(1 - \\braket{y}{\\Phi}\\right)^2$$\n",
+    "\n",
+    "This computes the normalised fidelity (similarity) of $\\ket{y}$ and $\\ket{\\Phi}$. We see that if $\\ket{y}$ and $\\ket{\\Phi}$ are equal, our cost function will equal 0, otherwise it will be greater than 0. Thus, we need to compute this cost function with our quantum hardware, coupled with classical minimising algorithms. \n",
+    "\n",
+    "Breaking down the cost function further, we see that it boils down to computing the following inner products, assuming real states:\n",
+    "\n",
+    "$$\\braket{y}{\\Phi} = a\\frac{C_2}{C_1} \\braket{y}{x} + \\frac{b}{C_1}\\braket{y}{1}$$\n",
+    "\n",
+    "### Computing Inner Products on a Quantum Computer\n",
+    "\n",
+    "It is clear we now need a quantum algorithm for computing inner products. Let us go through the theory here, which will be translated to quantum hardware in a couple of sections. \n",
+    "\n",
+    "Firstly, assume we have a state:\n",
+    "\n",
+    "$$\\ket{\\phi} = \\frac{1}{\\sqrt{2}}(\\ket{0}\\ket{x} + \\ket{1}\\ket{y})$$\n",
+    "\n",
+    "where we want to find the inner product, $\\braket{x}{y}$. Applying a Hadamard gate on the first qubit, we find:\n",
+    "\n",
+    "\\begin{equation}\n",
+    "    \\ket{\\phi} = \\frac{1}{2}(\\ket{0}(\\ket{x}+\\ket{y}) + \\ket{1}(\\ket{x}-\\ket{y}))\n",
+    "\\end{equation}\n",
+    "\n",
+    "This means that the probability to measure the first qubit as $\\ket{0}$ in the computational basis equals:\n",
+    "\n",
+    "\\begin{equation}\n",
+    "    P(0) = \\frac{1}{2}(1+Re[\\braket{x}{y}])\n",
+    "\\end{equation}\n",
+    "\n",
+    "This follows because:\n",
+    "\n",
+    "\\begin{align}\n",
+    "    P(0) &= |\\bra{0}\\otimes\\mathbb{1}\\ket{\\phi}|^2 \\\\\n",
+    "    &= \\frac{1}{4}\\braket{x+y}{x+y} \\\\\n",
+    "    &= \\frac{1}{4}(\\braket{x}{x}+\\braket{x}{y}+\\braket{y}{x}+\\braket{y}{y}) \\\\\n",
```

Comment:
Similarly for this. From this step it looks like |x + y> == |x> + |y>? It's not clear to me why that's the case.

Reply:
This is a good question as I should be more clear here. I am applying using the state $|\phi>$ after it has been acted on by the Hadamard gate on the first qubit. The |x + y> == |x> + |y> notation was a simplification, but I will get rid of it now as I see that it can be confusing. I have edited and recommitted. Hopefully it is a bit clearer now?

Reply:
Yes this is much clearer, thank you